components/keyrings/common/config/config_reader.h

```diff
@@ -76,3 +76,5 @@ class Config_reader {
 }  // namespace keyring_common
 
 #endif  // !CONFIG_READER_INCLUDED
+
+#include "config_reader.cc"
```

Comment:
`.h` file includes `.cc` file. Why???

Reply:
Maybe I should rename it to ".ic". This is required because we want to use LogComponentErr in the .cc file instead of fprintf. LogComponentErr needs a defined component name, and the keyring_common library doesn't know that. So I had to move the function implementations into the header file, and compile it in both keyring_file and keyring_kmip separately.